### DISCUSSION TITLE
Refetching data in getServerSideProps using apollo client.

### DISCUSSION THREAD

**Discussion by yassinebridi** [vercel/next.js#15430] [Help]
I'm using getServerSideProps with apollo client using this configuration:

[with-apollo](https://github.com/vercel/next.js/tree/canary/examples/with-apollo) 

But i need to update the page(refetch the query) when mutating, sadly it doesn't seem to work.

I don't know if this is th expected behavior from getServerSideProps or not. If so, should i just use getInitialProps instead.

**Comment by MorganeLecurieux** [comment]
Did you find an answer to your question ?

**Reply by yassinebridi** [reply]
Nope, still nothing.

**Reply by pureLifeWater101** [reply]
still no answer?

**Reply by kulame** [reply]
any update? i meet the same question

**Comment by jjorissen52** [comment]
I don't know anything about Apollo, but in general you can store the result of your getInitialProps query with a useState and update the state within the component whenever you perform a mutation.

You can always do a router.replace(router.asPath) to trigger a server-side render if you want to go that route.
